test/acceptance/libp2p/host/peer/tick_counter.hpp

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_TEST_ACCEPTANCE_LIBP2P_HOST_PEER_TICK_COUNTER_HPP
+#define KAGOME_TEST_ACCEPTANCE_LIBP2P_HOST_PEER_TICK_COUNTER_HPP
+
+#include <atomic>
+
+#include <gtest/gtest.h>
+
+/**
+ * @class TickCounter helper class which ensures that
+ * number of interactions is correct
+ */
+struct TickCounter {
+  TickCounter(size_t client_index, size_t server_index, uint32_t times)
+      : ci{client_index},
+        si{server_index},
+        required_count{times},
+        ticks_count{0u} {}
+
+  void tick() {
+    ++ticks_count;
+  }
+
+  void checkTicksCount() const {
+    auto actual_count = ticks_count.load();
+
+    if (required_count == actual_count) {
+      return;
+    }
+
+    ASSERT_EQ(actual_count, required_count)
```

Comment:
this will never be failing, because previous comparison returns

Reply:
It would pass to ASSERT_EQ if counts were not equal, but yes it is useless, forgot to remove.